### PR TITLE
GraphEmbed

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mtekman** [comment]
it needs a shed.yaml, otherwise it works

**Comment by mtekman** [comment]
can anyone debug the issue? All tests pass fine on three of my local machines.

**Comment by bgruening** [comment]
```
2019-12-07 14:13:15,826 INFO  [test_driver] Application galaxy stopped.
Testing complete. HTML report is in "/home/bag/projects/code/tools-iuc/tool_test_output.html".
There were problems with 2 test(s) - out of 2 test(s) executed. See /home/bag/projects/code/tools-iuc/tool_test_output.html for detailed breakdown.
graphembed[0]: failed
graphembed[1]: failed
```

Same error on my local setup then on travis.

**Comment by mtekman** [comment]
I'm at a loss on how to debug this. 

@bgruening what version of planemo are you using? I am getting passing tests on both `planemo, version 0.55.0` and `planemo, version 0.62.1`

**Comment by bgruening** [comment]
```
bag@bag:~/projects/code/tools-iuc$ planemo --version
planemo, version 0.62.1
```

File: tools/graphembed/graphembed.xml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+<tool id="graphembed" name="GraphEmbed" version="@VERSION@.0" >
+    <description>Compute a 2D embedding of a data matrix given supervised class information.</description>
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: remove the dot?

File: tools/graphembed/graphembed.xml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ '$advanced.normalisation'
+ '$advanced.feature_selection'
+ --cmap_name '$advanced.cmap_name'
+ --min_threshold '$advanced.min_threshold'
+ --max_threshold '$advanced.max_threshold'
+ --random_state '$advanced.random_state'
+#end if
+ --do_not_add_timestamp
+ --figure_size 15
+ --verbose
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: is verbose needed?

File: tools/graphembed/graphembed.xml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+ -o OUTPUT &&
+
+ cp OUTPUT/2D_coords.txt '$out_coords' &&
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: can you not remove this and use __from_working_dir__?

File: tools/graphembed/graphembed.xml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+ cp OUTPUT/classes.txt '$out_target' &&
+ cp OUTPUT/img_1_clean.pdf '$out_clean' &&
+ cp OUTPUT/img_2_links.pdf '$out_links' &&
+ ## cp OUTPUT/img_3_edge.pdf '$out_edge' &&
+ cp OUTPUT/img_4_hull.pdf '$out_hull' &&
+ cp OUTPUT/img_5_hull_link.pdf '$out_hulllink' &&
+ 
+ cat OUTPUT/log 2>/dev/stderr
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: not stdout?

File: tools/graphembed/graphembed.xml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+ ]]>
+    </command>
+
+    <inputs>
+        <param argument="-i" type="data" format="tabular"
+               label="Input Data File" />
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: how should this look like?

File: tools/graphembed/graphembed.xml [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        <param argument="-t" type="data" format="tabular"
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: How does this file need to look like?

File: tools/graphembed/graphembed.xml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        <param argument="-d" type="integer" value="1" min="0"
+               label="Number of links towards denser neighbours with a different class" />
+        <param argument="-z" type="integer" value="10" min="0"
+               label="Number of nearest neighbours to limit the horizon of where to
+                      search for denser neighbours of a different class" />
+        <param argument="-l" type="integer" value="0"
+               label="Number of mutual nearest neighbours that define outlier instances." />
+        <conditional name="advanced" >
+            <param name="option" type="select" label="Use Default Parameters?">
+                <option value="yes" selected="true" >Yes</option>
```
Comment (bgruening) [inline tools/graphembed/graphembed.xml]: Should the logic not be reverted. If yes = default parameters and then __no__ new options?

File: tools/graphembed/graphembed.xml (hunk 4 above)
Comment (mtekman) [reply tools/graphembed/graphembed.xml]: Well... piping to stdout will mean that the dataset in the history will expand to a large box of largely meaningless data. Whereas piping to stderr hides this and means only developers see it when debugging

Comment (mtekman) [reply tools/graphembed/graphembed.xml]: I decided to include the log as its own standalone file

**Pull request merged**